packages/mobile/src/networkInfo/saga.ts

```diff
@@ -34,6 +46,41 @@ function* subscribeToNetworkStatus() {
   }
 }
 
+function* fetchUserLocationData() {
+  let userLocationData: UserLocationData
+  try {
+    const response: Response = yield fetchWithTimeout(
+      networkConfig.fetchUserLocationDataUrl,
+      null,
+      FETCH_TIMEOUT_DURATION
+    )
+
+    userLocationData = yield response.json()
+
+    if (!response.ok) {
+      throw new Error(`IP address fetch failed. Error: ${JSON.stringify(userLocationData)}`)
+    }
+  } catch (error) {
+    Logger.error(`${TAG}:fetchUserLocationData`, error.message)
+    // If endpoint fails then use country code to determine location
+    const countryCallingCode: string | null = yield select(defaultCountryCodeSelector)
+    const countryCodeAlpha2 = countryCallingCode
+      ? getRegionCodeFromCountryCode(countryCallingCode)
+      : null
+    let ipAddress: string | null
+    try {
+      ipAddress = yield getIpAddress()
```

Comment:
Though `yield fnReturningPromise(arg)` work, we should use `yield call(fnReturningPromise, arg)` so it's easier to test. See https://redux-saga.js.org/docs/basics/DeclarativeEffects/